arangod/Replication2/StateMachines/Document/DocumentCore.cpp

```diff
@@ -32,10 +32,12 @@
 using namespace arangodb::replication2::replicated_state::document;
 
 DocumentCore::DocumentCore(
-    GlobalLogIdentifier gid, DocumentCoreParameters coreParameters,
+    TRI_vocbase_t& vocbase, GlobalLogIdentifier gid,
+    DocumentCoreParameters coreParameters,
     std::shared_ptr<IDocumentStateHandlersFactory> const& handlersFactory,
     LoggerContext loggerContext)
     : loggerContext(std::move(loggerContext)),
+      _vocbase(vocbase),
```

Comment:
How do we ensure that the vocbase lives long enough? I suppose we need a DatabaseGuard here.

Reply:
The vocbase strictly outlives all replicated states, in the sense that `resign` is called on all of them before the vocbase is destroyed.

Reply:
Apart from what Lars mentioned, a `DatabaseGuard` would still rely on `DatabaseFeature::useDatabase`, which is what causes the issues when looking up by name. Because Tobias decided to use a `vocbase` pointer, we'll be able to access it until the resignation of the replicated state, even though ''others" can no longer lookup the database.